### DISCUSSION TITLE
Lower # of CPU cores vs. Total Optical Drives

### DISCUSSION THREAD

**Discussion by ben-barnard** [automatic-ripping-machine/automatic-ripping-machine#760] [Q&A]
I'm sure I'm searching incorrectly, and this has been discussed but: I have an old Celeron system that can rip DVDs and music just fine, but only has two cores (G3900). I have connected 10+ optical drives and would like to use all of them. Let's say I input ten disks - is it possible for the software to queue the optical drives when there are no threads available (similar to how the transcoding works)? I don't care about overall ripping speed, but I do need the ability to swap out all disks at once (I'd like to just throw a bunch in each time I walk by). If it already works like this, I apologize.

When we define how many cores were are granting access to post-install, does a number of threads lower than the number of total optical drives available mean that it will split them across all of the disks inserted and process them immediately?

**Comment by Heath3827** [comment]
I don't have an exact answer for your use case (and given the age of your question, I'm sure you've figured something out), however I have been trying to work out using one of my own older machines and have some learnings for the next person to find this thread.

I have tried a CoreDuo, a Gen1 i3 and a Gen2 i5 all with similar results trying to process discs in 4 drives. In each case the disc is read and spat back out, then for the next 2 days the machine seems to have done nothing, however the allocated processor cores are maxed out with as much as 16 instances of HandBrake running and churning doing whatever magic incantations the developers have set up. 

In short, yes your Celery processor should be able to do the job, but expect each additional drive to increase the time greatly before the next disc can be processed. For practicality, I believe consideration needs to  given to allowing a core or 2 for each drive you plan to process. Since I have 3 machines with 2 or 4 threads each, I should realistically think of having 1 drive per machine or live with waiting 2 to 3 days (or a week for BluRay) between inserting discs (at least until I can obtain a newer machine with more cores or much faster processing abilities).
